File: backend/src/main/java/com/bookstore/entity/Purchase.java
package com.bookstore.entity;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Entity
@Table(name = "purchase_detail")
@EntityListeners(AuditingEntityListener.class)
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Purchase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 255)
    private String name;

    @Column(length = 20)
    private String phone;

    @Column(length = 500)
    private String books;

    @Column(nullable = false)
    private Integer quantity;

    @Column(name = "total_price", nullable = false, precision = 10, scale = 2)
    private Double totalPrice;

    @CreatedDate
    @Column(name = "date_purchased", updatable = false)
    private LocalDateTime datePurchased;
}

File: frontend/src/types/purchase.ts
export interface Purchase {
  id: number;
  name: string;
  phone: string;
  books: string;
  quantity: number;
  totalPrice: number;
  datePurchased: string;
}

export interface PurchaseRequest {
  name: string;
  phone: string;
  books: string;
  quantity: number;
  totalPrice: number;
}
